syntax = "proto3";
// Rectangle Area Calculator Definition
service NumberAdder{
	// computes area of a rectangle
	rpc add(stream TwoNumbers) returns (stream OneNumber) { }
	rpc add_oneshot(TwoNumbers) returns (OneNumber) { }
	rpc add_stream_single(stream OneNumber) returns (OneNumber) {}
}

message TwoNumbers {
	sint32 lhs = 1;
	sint32 rhs = 2;
}

message OneNumber {
	sint32 result = 1;
}
